lib/actions/field-trip.js

```diff
@@ -249,17 +406,355 @@ function planInbound (request) {
+
+      // set the pending amount of requests to be 2 so that there will always
+      // seem to be potentially additional requests that have to be made. If
+      // there aren't after making this next request, the pending amount will
+      // be set to 0. This needs to happen after the routingQuery so the search
+      // is defined.
+      dispatch(setPendingRequests({ searchId, pending: 2 }))
+
+      // obtain trip hashes from OTP Index API
+      await getTripHashesFromActiveItineraries()
+
+      // check trip validity and calculate itinerary capacity
+      const {
+        assignedItinerariesByResponse,
+        remainingGroupSize,
+        tripsToBanInSubsequentSearches
+      } = checkValidityAndCapacity(
+        getState(),
+        request
+      )
+
+      // always update itineraries on each itineration
+      dispatch(setActiveItineraries({
+        assignedItinerariesByResponse,
+        searchId
+      }))
+
+      if (remainingGroupSize <= 0) {
+        // All members of the field trip group have been assigned!
+        shouldContinueSearching = false
+        doFieldTripPlanRequestCleanup(dispatch, searchId)
+      } else {
+        // Not enough acceptable itineraries have been generated. Request more.
+
+        // Update banned trips
+        tripsToBanInSubsequentSearches.forEach(tripId => {
+          bannedTrips.add(tripId)
+        })
+        dispatch(setQueryParam({ bannedTrips: [...bannedTrips].join(',') }))
+      }
+    }
+  }
+}
+
+/**
+ * Makes a request to get data about other planned field trips happening on a
+ * particular date.
+ */
+async function getTripIdsForTravelDate (dispatch, state, request) {
+  const {datastoreUrl} = state.otp.config
+  const {sessionId} = state.callTaker.session
+  const formattedTravelDate = getFormattedRequestTravelDate(
+    request,
+    FIELD_TRIP_DATE_FORMAT
+  )
+  const params = {
+    date: formattedTravelDate,
+    sessionId
+  }
+
+  const res = await fetch(
+    `${datastoreUrl}/fieldtrip/getTrips?${qs.stringify(params)}`
+  )
+  const fieldTrips = await res.json()
+
+  // add passengers and converted tripId to trips
+  const trips = []
+  fieldTrips.forEach(fieldTrip => {
+    fieldTrip.groupItineraries.forEach(groupItinerary => {
+      groupItinerary.trips.forEach(gtfsTrip => {
+        // tripIds still stored as 'agencyAndId' in DB, so convert them to
+        // be compatible with OTP responses
+        gtfsTrip.tripId = gtfsTrip.agencyAndId.replace('_', ':')
+        // Add passengers to each trip from group itinerary
+        gtfsTrip.passengers = groupItinerary.passengers
+        trips.push(gtfsTrip)
+      })
+    })
+  })
+  await dispatch(receivedTravelDateTrips(trips))
+}
+
+/**
+ * Makes the needed requests to the OTP index API to obtain semantic trip hashes
+ * for any tripIds in the active itineraries that haven't already been obtained
+ * from the OTP index API.
+ */
+function getTripHashesFromActiveItineraries () {
```

Comment:
Method name is slightly confusing. It sounds like the hashes are coming from the itin objects. Simple name change to `getMissingTripHashesForActiveItineraries` might be helpful.

Reply:
Changed in a4480936984a2a1cb2d9dd6f97adc4ecd3a9a12c